src/og/ImageCanvas.js

```diff
@@ -1,227 +1,237 @@
-     */
-    load(url, callback) {
-        var img = new Image();
-        var that = this;
-        img.onload = function () {
-            that.resize(img.width, img.height);
-            that._context.drawImage(img, 0, 0, img.width, img.height);
-            callback && callback(img);
-        };
-        img.src = url;
-    }
-
-    /**
-     * Open canvas image in the new window.
-     * @public
-     */
-    openImage() {
-        var img = this.getImage();
-        var dataUrl = img.src;
-        var windowContent = "<!DOCTYPE html>";
-        windowContent += "<html>";
-        windowContent += "<head><title>Print</title></head>";
-        windowContent += "<body>";
-        windowContent += '<img src="' + dataUrl + '">';
-        windowContent += "</body>";
-        windowContent += "</html>";
-        var printWin = window.open(
-            "",
-            "",
-            "width=" + img.width + "px ,height=" + img.height + "px"
-        );
-        printWin.document.open();
-        printWin.document.write(windowContent);
-        printWin.document.close();
-        printWin.focus();
-    }
-
-    destroy() {
-        this._canvas.width = 1;
-        this._canvas.height = 1;
-        this._canvas = null;
-        this._context = null;
-    }
-}
-
-export { ImageCanvas };
+"use strict";
+
+/**
+ * Usefull class for working with JS canvas object.
+ * @class
+ */
+class ImageCanvas {
+    /**
+     * @param {number} [width] - Canvas width. Default 256.
+     * @param {number} [height] - Canvas height. Default 256.
+     */
+    constructor(width = 256, height = 256) {
+        /**
+         * Canvas object.
+         * @protected
+         * @type {Object}
+         */
+        this._canvas = document.createElement("canvas");
+        this._canvas.width = width;
+        this._canvas.height = height;
+
+        /**
+         * Canvas context.
+         * @protected
+         * @type {Object}
+         */
+        this._context = this._canvas.getContext("2d", {
+            willReadFrequently: true
+        });
+    }
+
+    /**
+     * Returns canvas object.
+     * @public
+     * @returns {Object}
+     */
+    getCanvas() {
+        return this._canvas;
+    }
+
+    /**
+     * Returns canvas context pointer.
+     * @public
+     * @returns {Object}
+     */
+    getContext() {
+        return this._context;
+    }
+
+    /**
+     * Fills canvas RGBA with zeroes.
+     * @public
+     */
+    fillEmpty() {
+        var imgd = this._context.getImageData(0, 0, this._canvas.width, this._canvas.height);
+        var pixels = imgd.data;
+        for (var i = 0, n = pixels.length; i < n; i += 4) {
+            pixels[i] = pixels[i + 1] = pixels[i + 2] = pixels[i + 3] = 0;
+        }
+        this._context.putImageData(imgd, 0, 0);
+    }
+
+    /**
+     * Fills canvas RGBA with color.
+     * @public
+     */
+    fill(c) {
+        console.log('CANVAS FILL W/', color)

```

Comment:
clean up. Also, It's hard to review all module, might be github bug.

Reply:
Please, replace the method into the end of the class

Reply:
One question: why would you move the method to the end of the class, because at the moment it is located right after the fillEmpty() method, what seems homogeneous to me?

Reply:
Agree:)